### PR TITLE
LPS-126912 Migrate Commerce account roles to Portal account roles

### PR DESCRIPTION
**Pull request by drewbrokke**
This is an update for [LPS-126912](https://issues.liferay.com/browse/LPS-126912).<br/><br/>/cc @pei-jung @alee8888 @ChrisKian @patriciajeanperez @lr-leo

### CODE REVIEW

File: modules/apps/commerce/commerce-account-service/src/main/java/com/liferay/commerce/account/internal/upgrade/v9_1_0/CommerceAccountRoleUpgradeProcess.java [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+		_roleLocalService = roleLocalService;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		try (PreparedStatement preparedStatement1 = connection.prepareStatement(
+				StringBundler.concat(
+					"select distinct UserGroupRole.roleId from UserGroupRole ",
+					"inner join Role_ on UserGroupRole.roleId = Role_.roleId ",
+					"where Role_.type_ =", RoleConstants.TYPE_SITE));
```
Comment (drewbrokke) [inline modules/apps/commerce/commerce-account-service/src/main/java/com/liferay/commerce/account/internal/upgrade/v9_1_0/CommerceAccountRoleUpgradeProcess.java:62]: Could we narrow this query down further by selecting on the type of group? So only pick up roleIds from UserGroupRoles that are associated with an Account group?

File: modules/apps/commerce/commerce-account-service/src/main/java/com/liferay/commerce/account/internal/upgrade/v9_1_0/CommerceAccountRoleUpgradeProcess.java [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+	private boolean _hasAccountEntryGroup(long roleId) throws Exception {
+		try (PreparedStatement preparedStatement = connection.prepareStatement(
+				StringBundler.concat(
+					"select count(distinct UserGroupRole.groupId)",
+					"from UserGroupRole inner join Group_ on ",
+					"UserGroupRole.groupId = Group_.groupId where ",
+					"UserGroupRole.roleId = ", roleId, " and ",
+					"Group.classNameId =",
+					_classNameLocalService.getClassNameId(
+						AccountEntry.class)))) {
```
[the remark this answers is not in the document]
Comment (drewbrokke) [reply modules/apps/commerce/commerce-account-service/src/main/java/com/liferay/commerce/account/internal/upgrade/v9_1_0/CommerceAccountRoleUpgradeProcess.java:152]: It works. See `EditCommerceAccountUserMVCActionCommand#editUserRoles`. UserGroupRoles uses the account's `groupId`.

File: modules/apps/commerce/commerce-account-service/src/main/java/com/liferay/commerce/account/internal/upgrade/v9_1_0/CommerceAccountRoleUpgradeProcess.java [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+				resourcePermission.getPrimKeyId());
+			newResourcePermission.setRoleId(accountRole.getRoleId());
+			newResourcePermission.setOwnerId(resourcePermission.getOwnerId());
+			newResourcePermission.setActionIds(
+				resourcePermission.getActionIds());
+			newResourcePermission.setViewActionId(
+				resourcePermission.getViewActionId());
+
+			_resourcePermissionLocalService.addResourcePermission(
+				newResourcePermission);
```
Comment (drewbrokke) [inline modules/apps/commerce/commerce-account-service/src/main/java/com/liferay/commerce/account/internal/upgrade/v9_1_0/CommerceAccountRoleUpgradeProcess.java:137]: How to handle different ResourcePermission scopes:

1. COMPANY should not be present
2. GROUP: check `primKey` to see if it is the an account group ID, if so update the `roleId`
3. GROUP_TEMPLATE: always copy
4. INDIVIDUAL: always copy

File: modules/apps/commerce/commerce-account-service/src/main/java/com/liferay/commerce/account/internal/upgrade/v9_1_0/CommerceAccountRoleUpgradeProcess.java [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+			PreparedStatement preparedStatement2 =
+				AutoBatchPreparedStatementUtil.autoBatch(
+					connection.prepareStatement(
+						"update Role_ set type_ = " +
+							RoleConstants.TYPE_ACCOUNT + " where roleId = ?"));
+			PreparedStatement preparedStatement3 =
+				AutoBatchPreparedStatementUtil.autoBatch(
+					connection.prepareStatement(
+						"update UserGroupRole set roleId =  ?" +
+							" where roleId = ?"))) {
```
[the remark this answers is not in the document]
Comment (drewbrokke) [reply modules/apps/commerce/commerce-account-service/src/main/java/com/liferay/commerce/account/internal/upgrade/v9_1_0/CommerceAccountRoleUpgradeProcess.java:72]: Mixed:
- If the UserGroupRole is of a site type Group, leave it alone
- If the UserGroupRole is of an account type Group, then change roleId to the created account roleId?

[the remark this answers is not in the document]
Comment (pei-jung) [reply modules/apps/commerce/commerce-account-service/src/main/java/com/liferay/commerce/account/internal/upgrade/v9_1_0/CommerceAccountRoleUpgradeProcess.java:72]: Add additional condition to only run the update when the group is an account's group

**Comment by pei-jung** [comment]
See https://github.com/liferay-usm/liferay-portal/pull/264.

**Pull request closed without merging**